modules/@apostrophecms/doc-type/index.js

```diff
@@ -94,28 +94,50 @@ module.exports = {
   handlers(self) {
     return {
       beforeSave: {
-        async updateBacklinks(req, doc) {
+        async updateCacheField(req, doc) {
           const relatedDocsIds = self.getRelatedDocsIds(req, doc);
 
-          // Remove all references to the doc
+          // - Remove current doc reference from docs that include it
+          // - Update these docs' cache field
           await self.apos.doc.db.updateMany({
             relatedReverseIds: { $in: [ doc.aposDocId ] },
             aposLocale: { $in: [ doc.aposLocale, null ] }
           }, {
-            $pull: { relatedReverseIds: doc.aposDocId }
+            $pull: { relatedReverseIds: doc.aposDocId },
+            $set: { cacheInvalidatedAt: doc.updatedAt }
           });
 
-          if (!relatedDocsIds.length) {
-            return;
+          if (relatedDocsIds.length) {
+            // - Add current doc reference to related docs
+            // - Update related docs' cache field
+            await self.apos.doc.db.updateMany({
+              aposDocId: { $in: relatedDocsIds },
+              aposLocale: { $in: [ doc.aposLocale, null ] }
+            }, {
+              $push: { relatedReverseIds: doc.aposDocId },
+              $set: { cacheInvalidatedAt: doc.updatedAt }
+            });
           }
 
-          // Add doc reference to all related docs
-          await self.apos.doc.db.updateMany({
-            aposDocId: { $in: relatedDocsIds },
-            aposLocale: { $in: [ doc.aposLocale, null ] }
-          }, {
-            $push: { relatedReverseIds: doc.aposDocId }
-          });
+          if (doc.relatedReverseIds && doc.relatedReverseIds.length) {
+            // Update related reverse docs' cache field
+            await self.apos.doc.db.updateMany({
+              aposDocId: { $in: doc.relatedReverseIds },
+              aposLocale: { $in: [ doc.aposLocale, null ] }
+            }, {
+              $set: { cacheInvalidatedAt: doc.updatedAt }
+            });
+          }
+
+          if (doc._parentUrl) {
+            // Update piece index page's cache field
+            await self.apos.doc.db.updateOne({
+              slug: doc._parentUrl,
```

Comment:
This will not work in the general case, the URL is not guaranteed to be identical to the slug, consider sitewide prefixes for instance.

It's OK to use this mechanism but you should do it by adding a `doc._parentPage` property to go along with `_parentUrl` so this can be done correctly (make this change in `addUrlsToPieces`).

Reply:
Sitewide prefixes like `en/` in `somesite.com/en/some-slug`?

I'll add `piece._parentPage = parentPage.slug;` in `addUrlsToPieces ` then.